### ISSUE TITLE
azurerm_virtual_machine_scale_set does not support 'licenseType' property Azure Hybrid Use Benefit

### ISSUE THREAD

**Issue by deyjcode** [terraform-providers/terraform-provider-azurerm#1113]
### Terraform Version
Terraform v0.11.7
+ provider.azurerm v1.3.2
+ provider.random v1.2.0

### Affected Resource(s)
- azurerm_virtual_machine_scale_set

### Expected Behavior
Provide support for and parity with terraform AzureRM resource 'azurerm_virtual_machine'.

### Actual Behavior
Property 'license_type' is missing.

### Important Factoids
Refer to Azure's FAQ on Hybrid Use (https://docs.microsoft.com/en-us/azure/virtual-machines/windows/hybrid-use-benefit-licensing). 
Azure provides JSON example for Scale Sets to enable Hybrid Use via property 'licenseType'
```json
"virtualMachineProfile": {
    "storageProfile": {
        "osDisk": {
            "createOption": "FromImage"
        },
        "imageReference": {
            "publisher": "MicrosoftWindowsServer",
            "offer": "WindowsServer",
            "sku": "2016-Datacenter",
            "version": "latest"
        }
    },
    "licenseType": "Windows_Server",
    "osProfile": {
            "computerNamePrefix": "[parameters('vmssName')]",
            "adminUsername": "[parameters('adminUsername')]",
            "adminPassword": "[parameters('adminPassword')]"
    }
```

### References
Not sure if this resolves this below issue, but linking for visibility regardless.
https://github.com/Azure/terraform-azurerm-compute/issues/59

**Comment by whytoe** [comment]
This feature does have a bad downside in respects to Microsoft Licensing and you have to be very careful using it.  believe it will allocate a licence for 3 months or 1 year. (It is hard to get the correct answer from Microsoft Licensing)

If you have a Hybrid Use Benefit and it could lock all of your licensing up in Scale Sets potentially unless something has changed in the past few months.

**Comment by tombuildsstuff** [comment]
Fixed via #1245
